server/routes/visitors.ts

```diff
@@ -290,11 +295,22 @@ export default function routes(
       res,
     })
 
-    const { offenderNo } = req.session.visitSessionData.prisoner
+    const formValues = (req.flash('formValues')?.[0] as unknown as Record<string, string | string[]>) || {}
+
+    if (!Object.keys(formValues).length && req.session.visitSessionData.mainContact) {
```

Comment:
Maybe assign `req.session.visitSessionData` to a variable?

Reply:
..to make it easier to read.

Reply:
Done - in fact as it was repeated a lot I got carried away and did it throughout this file and the utils file.